distributed/scheduler.py

```diff
@@ -904,13 +904,15 @@ def remove_worker(self, comm=None, address=None, safe=False, close=True):
             del self.worker_comms[address]
             del self.ncores[address]
             self.workers.remove(address)
+            self.clear_ip_cache(self.worker_info[address]['name'])
```

Comment:
I think that you don't want name here.  I think that you want host.  Name is assigned aliases when they exist.

Reply:
Can we call `ensure_ip.clear_cache(self.worker_info[address]['host'])` here directly and avoid touching the comms layer and all of the `clear_address_cache` definitions?

Reply:
I think @mrocklin has a point. Also, it would be nice to add a comment here explaining why we're doing this.